app/Helpers/CollectionHelper.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+
+namespace App\Helpers;
+
+use Illuminate\Container\Container;
+use Illuminate\Pagination\LengthAwarePaginator;
+use Illuminate\Pagination\Paginator;
+use Illuminate\Support\Collection;
+
+/**
+ * Class CollectionHelper
+ * Laravel supports eloquent model pagination out of the box, but not custom collection.
+ * This helper provides an easy way to paginate any collections.
+ *
+ * @see https://sam-ngu.medium.com/laravel-how-to-paginate-collection-8cb4b281bc55
+ *
+ * @package App\Helpers
+ */
+class CollectionHelper
+{
+    public static function paginate(Collection $results, $pageSize)
```

Comment:
To check what you are returning is great to use Tinkerwell in some cases. 
Because there you can see directly the output of the last line of code you are dealing with, 
so you can see what class type is.